File: frontend/src/services/valuation.ts
import { api } from "./api";

export interface ValuationHolding {
  id: number;
  symbol: string;
  quantity: number;
  average_buy_price: number;
  invested_amount: number;
  current_price: number;
  current_value: number;
  profit_loss: number;
  profit_loss_percent: number;
}

export interface PortfolioSummary {
  total_invested: number;
  current_value: number;
  profit_loss: number;
  profit_loss_percent: number;
  holdings_count: number;
}

export interface PortfolioValuation {
  summary: PortfolioSummary;
  holdings: ValuationHolding[];
}

export async function getPortfolioValuation(): Promise<PortfolioValuation> {
  return api.get<PortfolioValuation>("/portfolio/valuation");
}

File: backend/app/schemas/valuation.py
from pydantic import BaseModel


class HoldingValuation(BaseModel):
    id: int
    symbol: str
    quantity: float
    average_buy_price: float
    invested_amount: float
    current_price: float
    current_value: float
    profit_loss: float
    profit_loss_percent: float


class PortfolioSummary(BaseModel):
    total_invested: float
    current_value: float
    profit_loss: float
    profit_loss_percent: float
    holdings_count: int


class PortfolioValuationResponse(BaseModel):
    summary: PortfolioSummary
    holdings: list[HoldingValuation]
